pyteal/ast/itxn.py

```diff
@@ -200,6 +206,178 @@ def SetFields(cls, fields: Dict[TxnField, Union[Expr, List[Expr]]]) -> Expr:
         fieldsToSet = [cls.SetField(field, value) for field, value in fields.items()]
         return Seq(fieldsToSet)
 
+    @classmethod
+    def MethodCall(
+        cls,
+        *,
+        app_id: Expr,
+        method_signature: str,
+        args: list[abi.BaseType | Expr | dict[TxnField, Expr | list[Expr]]],
+        extra_fields: dict[TxnField, Expr | list[Expr]] = {},
+    ) -> Expr:
+        """Adds an ABI method call transaction to the current inner transaction group.
+
+        :any:`InnerTxnBuilder.Begin` must be called before a MethodCall can be added.
+
+        Requires Teal version 6 or higher. This operation is only permitted in application mode.
+
+        Args:
+            app_id: An expression that evaluates to a `TealType.uint64` corresponding to the application being called.
+            method_signature: A string representing the method signature of the method we're calling. This is used to do
+                type checking on the arguments passed and to create the method selector passed as the first argument.
+            args: A list of arguments to pass to the application. The values in this list depend on the kind of argument you wish to pass:
+
+                - For basic ABI arguments (not Reference or Transaction types): The ABI type is passed directly and encoded as part of transaction preparation. The type passed _MUST_ match the type specified in the `method_signature` passed.
```

Comment:
nit: the `_MUST_` in this line doesn't get formatted at all in the docs. Bold & italics are described here: https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#inline-markup-and-special-characters-e-g-bold-italic-verbatim